### PR TITLE
feat: [P4ADEV-206]  BE Mock for Transaction detail API 

### PR DESCRIPTION
**Pull request by AnisLucidi01**
… for transaction detail.

#### List of Changes

Minor changes to transaction detail interface to better accomodate the BE Mock. Integrated the mock API.

#### Motivation and Context

This change is required so that we have a BE Mock working, to ease the future transition from mock BE to actual BE.
#### How Has This Been Tested?

Ran jest testing and coverage, visual testing. 

#### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

#### Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: src/App.tsx [hunk 1]
```diff
@@ -48,6 +48,7 @@ const router = createBrowserRouter([
       {
         path: ArcRoutes.TRANSACTION,
         element: <TransactionRoute />,
+        loader: () => utils.apiClient.transactions.getTransactionDetails('1'),
```
Comment (stratoivandiluccio) [inline src/App.tsx:51]: We need to pass the actual id, not an hardcoded one.

File: src/components/Transactions/TransactionDetail.tsx [hunk 2]
```diff
@@ -38,7 +38,7 @@ export default function TransactionDetail({
       <Stack spacing={2} mt={3} width={'100%'}>
         <Box>
           <Typography>
-            {t('app.transactionDetail.createdOn') + ' ' + transactionData.dateTime.toDateString()}
+            {t('app.transactionDetail.createdOn') + ' ' + transactionData.dateTime}
```
Comment (vito80ba) [inline src/components/Transactions/TransactionDetail.tsx:41]: An use of "Template literal" is appreciated :-)

File: src/components/Transactions/TransactionDetail.tsx [hunk 3]
```diff
@@ -38,7 +38,7 @@ export default function TransactionDetail({
       <Stack spacing={2} mt={3} width={'100%'}>
         <Box>
           <Typography>
-            {t('app.transactionDetail.createdOn') + ' ' + transactionData.dateTime.toDateString()}
+            {`${t('app.transactionDetail.createdOn')} ${transactionData.dateTime}`}
```
Comment (vito80ba) [inline src/components/Transactions/TransactionDetail.tsx:41]: Screenshot 2024-05-08 alle 15 13 01
please, use [this function](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/DateTimeFormat) to have a better render of date

File: src/components/Transactions/TransactionDetail.tsx [hunk 4]
```diff
@@ -134,9 +134,7 @@ export default function TransactionDetail({
                   <Typography color={theme.palette.text.secondary}>
                     {t('app.transactionDetail.dateAndTime')}
                   </Typography>
-                  <Typography fontWeight={600}>
-                    {transactionData.dateTime.toDateString()}
-                  </Typography>
+                  <Typography fontWeight={600}>{transactionData.dateTime}</Typography>
```
Comment (vito80ba) [inline src/components/Transactions/TransactionDetail.tsx:137]: The same as above

**Review by vito80ba — CHANGES_REQUESTED** [review]

File: src/components/Transactions/TransactionDetail.tsx (hunk 4 above)
Comment (AnisLucidi01) [reply src/components/Transactions/TransactionDetail.tsx:137]: Small problem.. use this, you need to have a date object, to make a date object, you need to have a valid string (so the string that is returned from the be mock, which is "string" doesn't work, so it throws an error). What should I do? Implement it with a fake date (maybe in the converter util), or leave it as it is?

Comment (vito80ba) [reply src/components/Transactions/TransactionDetail.tsx:137]: If you have a string you can cast in this way
```
const serviceValue = "21/03/2024".split("/")
const dateformat = new Date(serviceValue[2], serviceValue[1], serviceValue[0]);
```
then
```
const dateHumnaFriendly = new Intl.DateTimeFormat(LANG).{
    dateStyle: 'short',
    timeStyle: 'short'
  }.format(dateformat)
```

Comment (AnisLucidi01) [reply src/components/Transactions/TransactionDetail.tsx:137]: The problem is that the BE returns a string with "string" as a value, which is not accepted by the new Date constructor, even if I split it, it'd still give me an empty value, which isn't liked by the constructor.

**Review by vito80ba — APPROVED** [review]

**Pull request merged**